### PR TITLE
Nf 397 split businessname from address

### PR DESCRIPTION
**Pull request by rorticus0502**

### CODE REVIEW

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/navigation/CreateNavigator.scala [hunk 1]
```diff
@@ -76,6 +77,12 @@ class CreateNavigator @Inject() ()
       enterImporterEori,
       importerEoriNumberAnswered
     ),
+    P(
+      ImporterBusinessNamePage,
+      makeclaim.routes.ImporterBusinessNameController.onPageLoad,
+      always,
```
Comment (SteveSugden) [inline app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/navigation/CreateNavigator.scala:83]: I'm not sure that "always" is right here.  Should it not be "isRepresentative"

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/navigation/CreateNavigator.scala [hunk 2]
```diff
@@ -90,49 +97,53 @@ class CreateNavigator @Inject() ()
...
+  val itemNumbers          = "item-numbers"
+  val claimReason          = "claim-reason"
+  val dutyTypes            = "duty-types"
+  val dutyCustoms          = "duty-customs"
+  val dutyVAT              = "duty-vat"
+  val dutyOther            = "duty-other"
+  val dutySummary          = "duty-summary"
+  val uploadSummary        = "uploaded-files"
+  val contactDetails       = "contact-details"
+  val businessName         = "enter-business-name"
```
Comment (SteveSugden) [inline app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/navigation/CreateNavigator.scala:139]: The value of these labels doesn't really matter.  But "business-name" and "importer-business-name" would be more consistent.

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/views/makeclaim/BusinessNameView.scala.html [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            legend = "businessName.title",
+            hint = Some(Html(messages("businessName.hint")))
+        ) {
+
+            @inputText(
+                form = form,
+                id = "name",
+                name = "name",
+                label = "businessName.name.heading",
+                autocomplete = Some("name")
```
Comment (SteveSugden) [inline app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/views/makeclaim/BusinessNameView.scala.html:51]: Not sure about that autocomplete.   It was `autocomplete = Some("off")` on the address form - not sure why - surely `None` would be the setting to turn it off

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/views/makeclaim/CreateClaimSummary.scala.html [hunk 4]
```diff
@@ -234,6 +234,25 @@
...
+            ),
+            value = Value(
+                content = Text(answers.businessName.map(businessName => businessName.name).getOrElse(""))
+            ),
+            actions = if(!withActions) None else  Some(Actions(
+                items = Seq(
+                    actionItem(
+                        href = makeclaim.routes.CheckYourAnswersController.onChange(CreatePageNames.businessName).url,
+                        content = HtmlContent(linkContent(changeText(answers.businessName))),
+                        visuallyHiddenText = Some(messages("check_answers.yourDetails.contactDetails.accessible"))
```
Comment (SteveSugden) [inline app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/views/makeclaim/CreateClaimSummary.scala.html:251]: Do you need some new accessible text for business name?

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/views/makeclaim/CreateClaimSummary.scala.html [hunk 5]
```diff
@@ -360,6 +378,26 @@
...
+                    ),
+                    value = Value(
+                        content = Text(answers.importerBusinessName.map(importerBusinessName => importerBusinessName.name).getOrElse(""))
+                    ),
+                    actions = if(!withActions) None else  Some(Actions(
+                        items = Seq(
+                            actionItem(
+                                href = makeclaim.routes.CheckYourAnswersController.onChange(CreatePageNames.importerBusinessName).url,
+                                content = HtmlContent(linkContent(changeText(answers.importerBusinessName))),
+                                visuallyHiddenText = Some(messages("check_answers.importer.contactDetails.accessible"))
```
Comment (SteveSugden) [inline app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/views/makeclaim/CreateClaimSummary.scala.html:395]: Do you need some new accessible text for importer's business name?

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/views/makeclaim/ImporterBusinessNameView.scala.html [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+            legend = "importer.businessName.title",
+            hint = Some(Html(messages("businessName.hint")))
+        ) {
+
+            @inputText(
+                form = form,
+                id = "name",
+                name = "name",
+                label = "importer.businessName.name.heading",
+                autocomplete = Some("name")
```
Comment (SteveSugden) [inline app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/views/makeclaim/ImporterBusinessNameView.scala.html:51]: Same comment re: autocomplete

File: conf/messages [hunk 7]
```diff
@@ -153,11 +153,14 @@ contactDetails.emailAddress.error.length = Email must be 85 characters or fewer
 contactDetails.telephoneNumber.heading = UK telephone number
 contactDetails.telephoneNumber.error.length = Telephone number must be between 9 and 32 characters
 
+businessName.title = What is your business name
+businessName.hint =
+businessName.name.heading =
```
Comment (SteveSugden) [inline conf/messages:158]: Unused or missing message definitions?

File: conf/messages [hunk 8]
```diff
@@ -175,6 +178,12 @@ address.postcode.error.invalid = Enter a real postcode
 importer.has.eori.title = Does the importer have an EORI number?
 importer.has.eori.required = Select yes if the importer has an EORI number
 
+importer.businessName.title = What is the name of the importers business
+importer.businessName.hint =
+importer.businessName.name.heading =
```
Comment (SteveSugden) [inline conf/messages:183]: Unused or missing message definitions?

File: test/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/views/makeclaim/CheckYourAnswersViewSpec.scala [hunk 9]
```diff
@@ -221,7 +221,7 @@ class CheckYourAnswersViewSpec extends UnitViewSpec with TestData {
 
         addressDetailsRow must haveSummaryKey(messages("check_answers.yourDetails.yourAddress"))
         addressDetailsRow must haveSummaryValue(
-          s"${addressAnswer.name} ${addressAnswer.addressLine1} ${addressAnswer.addressLine2.get} ${addressAnswer.city} ${addressAnswer.postCode}"
+          s"${addressAnswer.addressLine1} ${addressAnswer.addressLine2.get} ${addressAnswer.city} ${addressAnswer.postCode}"
         )
```
Comment (SteveSugden) [inline test/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/views/makeclaim/CheckYourAnswersViewSpec.scala:226]: Can you add some tests for two new rows on CYA?   I think all rows are currently tested

**Review by SteveSugden — CHANGES_REQUESTED** [review]
A few comments, mostly minor suggestions.

The only real concern I have is the logic in the CreateNavigator

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/navigation/CreateNavigator.scala (hunk 1 above)
Comment (rorticus0502) [reply app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/navigation/CreateNavigator.scala:83]: good spot - thank you

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/navigation/CreateNavigator.scala (hunk 2 above)
Comment (rorticus0502) [reply app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/navigation/CreateNavigator.scala:139]: I just used the urls provided in the jira ticket as these labels.

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/views/makeclaim/CreateClaimSummary.scala.html (hunk 5 above)
Comment (rorticus0502) [reply app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/views/makeclaim/CreateClaimSummary.scala.html:395]: Yes - will correct now

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/views/makeclaim/CreateClaimSummary.scala.html (hunk 4 above)
Comment (rorticus0502) [reply app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/views/makeclaim/CreateClaimSummary.scala.html:251]: Yep - will correct now

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/views/makeclaim/BusinessNameView.scala.html (hunk 3 above)
Comment (rorticus0502) [reply app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/views/makeclaim/BusinessNameView.scala.html]: None works - so I have changed it to that

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/views/makeclaim/ImporterBusinessNameView.scala.html (hunk 6 above)
Comment (rorticus0502) [reply app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/views/makeclaim/ImporterBusinessNameView.scala.html]: checked that None works and it does - changed it as requested

File: conf/messages (hunk 7 above)
Comment (rorticus0502) [reply conf/messages]: removed

File: conf/messages (hunk 8 above)
Comment (rorticus0502) [reply conf/messages]: removed

File: test/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/views/makeclaim/CheckYourAnswersViewSpec.scala (hunk 9 above)
Comment (rorticus0502) [reply test/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/views/makeclaim/CheckYourAnswersViewSpec.scala:242]: tests added - shouldnt have missed that - but they are there now

**Review by SteveSugden — APPROVED** [review]
LGTM.   The "Viewed" buttons work really well on PRs - must use them more often.

**Pull request merged**